server/src/main/java/com/skillmagnet/Lesson/Lesson.java

```diff
@@ -0,0 +1,39 @@
+package com.skillmagnet.Lesson;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.skillmagnet.Course.Course;
+import jakarta.persistence.*;
+import lombok.*;
+
+@Getter
+@Setter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
```

Comment:
There's no proper reason - more just personal taste choice lol. The `@Data` does include all of that stuff, but also adds toString(), toEquals() and some other stuff that I'm not actually implementing. While functionally it changes absolutely nothing, I just like seeing all the different effects on my class being very verbose and accurate to what I want rather than lumping in bunch of stuff with `@Data`.

TLDR: I prefer extremely verbose code at the tradeoff of taking more lines. Completely debatable